### PR TITLE
Re21

### PR DESCRIPTION
**Pull request by fidransky**
Specs for WS API (see #21).

### CODE REVIEW

File: discussment-ws/docs/discussment-service-v1.yaml [hunk 1]
```diff
@@ -0,0 +1,583 @@
...
+      level:
+        description: Nesting level of the post. Starts from 0, increases by 1 for every reply level.
+        type: integer
+        required: true
+      createdBy:
+        description: Username of the user who authored the post.
+        type: string
+        required: true
+      createdAt:
+        description: Datetime when the post was created in UTC timezone.
```
Comment (danekja) [inline discussment-ws/docs/discussment-service-v1.yaml:160]: Description should give explicit example of formatting. Datetimes are always trouble when de/serializing.

File: discussment-ws/docs/discussment-service-v1.yaml [hunk 2]
```diff
@@ -0,0 +1,583 @@
...
+title: Discussment web service
+version: v1
+baseUri: /{version}
+mediaType: application/json
+securitySchemes:
+  basic:
+    description: JWT-based authentication.
+    type: Basic Authentication
+securedBy: [ basic ]
+types:
```
Comment (danekja) [inline discussment-ws/docs/discussment-service-v1.yaml:11]: RAML types can be in separate file and only included here. Makes the API document more readable when editing. For reading full API html format is better imho.

File: discussment-ws/docs/discussment-service-v1.yaml [hunk 3]
```diff
@@ -0,0 +1,583 @@
+#%RAML 1.0
+title: Discussment web service
+version: v1
+baseUri: /{version}
+mediaType: application/json
+securitySchemes:
```
Comment (danekja) [inline discussment-ws/docs/discussment-service-v1.yaml:6]: Security schemes can be in separate file and only included here. Makes the API document more readable when editing. For reading full API html format is better imho.

File: discussment-ws/docs/discussment-service-v1.yaml [hunk 4]
```diff
@@ -0,0 +1,583 @@
...
+      401:
+        description: User is not authenticated.
+      403:
+        description: User is not authorized to read all categories.
+  post:
+    displayName: createCategory
+    description: Creates a new category.
+    body: CreateCategoryDTO
+    responses:
+      200:
```
Comment (danekja) [inline discussment-ws/docs/discussment-service-v1.yaml:232]: Should be response code 201 Created

File: discussment-ws/docs/discussment-service-v1.yaml [hunk 5]
```diff
@@ -0,0 +1,583 @@
...
+          403:
+            description: User is not authorized to read topics in the category.
+          404:
+            description: Category with this id does not exist.
+      post:
+        displayName: createTopic
+        description: Creates a new topic in given category.
+        body: CreateTopicDTO
+        responses:
+          200:
```
Comment (danekja) [inline discussment-ws/docs/discussment-service-v1.yaml:289]: Should be response code 201 Created

File: discussment-ws/docs/discussment-service-v1.yaml [hunk 6]
```diff
@@ -0,0 +1,583 @@
...
+          403:
+            description: User is not authorized to read discussions in the topic.
+          404:
+            description: Topic with this id does not exist.
+      post:
+        displayName: createDiscussion
+        description: Creates a new discussion in given topic.
+        body: CreateDiscussionDTO
+        responses:
+          200:
```
Comment (danekja) [inline discussment-ws/docs/discussment-service-v1.yaml:350]: Should be response code 201 Created

File: discussment-ws/docs/discussment-service-v1.yaml (hunk 1 above)
Comment (fidransky) [reply discussment-ws/docs/discussment-service-v1.yaml:160]: Format is specified by RAML's `type: datetime-only`.

Comment (danekja) [reply discussment-ws/docs/discussment-service-v1.yaml]: When I read the documentation I dont know the format. Documentation needs to be explicit.

**Pull request merged**